src/actions/tamanu/prepare_upgrade.rs

```diff
@@ -0,0 +1,160 @@
+use std::{
+	collections::HashSet,
+	fs,
+	path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use itertools::Itertools;
+use miette::{bail, IntoDiagnostic, Result, WrapErr};
+use node_semver::Version;
+use regex::Regex;
+use tracing::info;
+
+use crate::actions::{
+	tamanu::{
+		download::{download, make_url},
+		find_package,
+	},
+	Context,
+};
+
+use super::{download::ServerKind, find_existing_version, find_tamanu, TamanuArgs};
+
+/// Setup new Tamanu instllation without taking down the existing one
+#[derive(Debug, Clone, Parser)]
+pub struct PrepareUpgrade {
+	/// Version to update to.
+	#[arg(value_name = "VERSION")]
+	pub version: Version,
+
+	/// Package to upgrade.
+	///
+	/// By default, this command looks for the most recent installed version of Tamanu.
+	/// If both central and facility servers are present and
+	/// configured, it will pick one arbitrarily.
+	#[arg(short, long)]
+	pub package: Option<String>,
```

Comment:
```suggestion
	pub package: Option<ServerKind>,
```

As `ServerKind` derives `ValueEnum`, it can be used directly in the args and clap will take care of parsing the value and erroring as appropriate.

Reply:
One problem I see with that is `ServerKind` includes the `Web` variant, which is not relevant for `prepare_upgrade`. Using `ServerKind` decreases the number of invalid input `run` needs to deal with, but doesn't decrease the number of branching I need to handle them. (In practice, making `package` `ServerKind` mean that I need to throw an error for `Web` when turning it into a package name like `central-server`.) I once thought of generalise `ServerKind` by having an enum with `Central` and `Facility` and another enum with `Web` and a variant referring to the first enum. I wasn't sure if it was worthwhile.

Reply:
The other advantage of valueenum is it hooks into help and completions for possiblevalues, so i'd say it's still worthwhile. Perhaps a ApiServerKind with a From impl to convert